### ISSUE TITLE
The json option does not work anymore.

### ISSUE THREAD

**Issue by nathanpeck** [request/request#1078]
According to the documentation:

>  json - sets body but to JSON representation of value and adds Content-type: application/json header. Additionally, parses the response body as JSON.

This property does not seem to work anymore.

Here is what I experienced:

1) I had an older project where my mikael/request dependency version was just specified as `"*"` in package.json
2) Recently I deleted the node_modules folder, did an `npm install` to reinstall all the dependencies, and it pulled in the latest version of request (2.42.0), over the older version that it had before.
3) After the NPM install I noticed that my project was no longer sending JSON bodies with its POST requests as it did before. Instead on the other end of the line the POST requests were coming through with empty bodies. Naturally this broke a lot of things.
4) I was able to get my code working again by changing from using the `json` property to the `body` property.

Example of what I had before which used to work in the older version of request, and which should still work according to the docs:

```js
    request(
      {
        url: 'REDACTED',
        method: 'POST',
        json: {
          foo: 'bar'
        }
      },
      function (err, response, body) {}
    );
```

The above example no longer worked, however changing it to this did work:

```js
   request(
      {
        url: 'REDACTED',
        method: 'POST',
        body: {
          foo: 'bar'
        }
      },
      function (err, response, body) {}
    );
```

In summary it appears that the `json` options longer functions, or has been deprecated.

If this is the case then it should be removed from the documentation in the README since it no longer works. However, if it is intended to work then this would be a bug.
